js/auth.js

```diff
@@ -0,0 +1,306 @@
+  }
+  checkLoginButton(); // 버튼 상태 체크
+};
+
+const SignupEmailFocusout = () => {
+  if (signupEmail.value === "") {
+    signupEmail.classList.add("empty_input");
+    signupEmailError.textContent = "이메일을 입력해주세요";
+    signupEmailError.classList.add("show");
+  } else if (!isValidEmail(signupEmail.value)) {
+    signupEmail.classList.add("empty_input");
+    signupEmailError.textContent = "잘못된 이메일 형식입니다";
+    signupEmailError.classList.add("show");
+  } else {
+    signupEmail.classList.remove("empty_input");
+    signupEmailError.classList.remove("show");
+  }
+  checkSignupButton();
+};
+
+const LoginPasswordFocusout = () => {
+  if (loginPassword.value === "") {
+    loginPassword.classList.add("empty_input");
+    passwordError.textContent = "비밀번호를 입력해주세요";
+    passwordError.classList.add("show");
+  } else if (loginPassword.value.length < 8) {
+    loginPassword.classList.add("empty_input");
+    passwordError.textContent = "비밀번호를 8자 이상 입력해주세요";
+    passwordError.classList.add("show");
+  } else {
+    loginPassword.classList.remove("empty_input");
+    passwordError.classList.remove("show");
+  }
+  checkLoginButton();
+};
+
+const SignupPasswordFocusout = () => {
+  if (signupPassword.value === "") {
+    signupPassword.classList.add("empty_input");
+    signupPasswordError.textContent = "비밀번호를 입력해주세요";
+    signupPasswordError.classList.add("show");
+  } else if (signupPassword.value.length < 8) {
+    signupPassword.classList.add("empty_input");
+    signupPasswordError.textContent = "비밀번호를 8자 이상 입력해주세요";
+    signupPasswordError.classList.add("show");
+  } else {
+    signupPassword.classList.remove("empty_input");
+    signupPasswordError.classList.remove("show");
+  }
+  checkSignupButton();
+};
+
+const NicknameFocusout = () => {
+  if (signupNickname.value === "") {
+    signupNickname.classList.add("empty_input");
+    nicknameError.textContent = "닉네임을 입력해주세요";
+    nicknameError.classList.add("show");
+  } else {
+    signupNickname.classList.remove("empty_input");
+    nicknameError.classList.remove("show");
+  }
+  checkSignupButton();
+};
+
+const PasswordCheckFocusout = () => {
+  if (signupPassword.value !== signupPasswordCheck.value) {
+    signupPasswordCheck.classList.add("empty_input");
+    signupPasswordCheckError.textContent = "비밀번호가 일치하지 않습니다";
+    signupPasswordCheckError.classList.add("show");
+  } else {
+    signupPasswordCheck.classList.remove("empty_input");
+    signupPasswordCheckError.classList.remove("show");
+  }
+  checkSignupButton();
+};
+
+// 버튼 상태 체크 함수
+const checkLoginButton = () => {
+  const hasEmailError = emailError.classList.contains("show");
+  const hasPasswordError = passwordError.classList.contains("show");
+  const isEmailEmpty = loginEmail.value === "";
+  const isPasswordEmpty = loginPassword.value === "";
+
+  if (isEmailEmpty || isPasswordEmpty || hasEmailError || hasPasswordError) {
+    loginButton.disabled = true;
+  } else {
+    loginButton.disabled = false;
+  }
+};
```

Comment:
💊 제안
아래처럼도 작성이 가능합니다!

```suggestion
const checkLoginButton = () => {
  const hasEmailError = emailError.classList.contains("show");
  const hasPasswordError = passwordError.classList.contains("show");
  const isEmailEmpty = loginEmail.value === "";
  const isPasswordEmpty = loginPassword.value === "";

  const isValid = isEmailEmpty && isPasswordEmpty && hasEmailError && hasPasswordError;
  loginButton.disabled = !isValid;
};
```

내부의 로직을 목적별로 분리하시면 가독성이 더 좋아집니다.
이렇게 목적별로 분리하면 checkLoginButton 내부에서 호출하는 함수명을 통해 어떤 값을 확인하는지 파악하기 쉬워집니다.

```suggestion
const isValidEmailInput = () => {...} // 실행시 boolean 값 반환
const isValidPasswordInput = () => {...} // 실행시 boolean 값 반환

const checkLoginButton = () => {
  const isValid = isValidEmailInput() && isValidPasswordInput();
  loginButton.disabled = !isValid;
};
```